### DISCUSSION TITLE
Unable to run tests in diferent assemblies un parallel

### DISCUSSION THREAD

**Discussion by ionmincu** [xunit/xunit#2809] [Question]
Please help. I am unable to make xunit run tests from multiple assemblies in parallel. I can only get it to one assembly at a time.
I tried diferent settings / arguments, searching for `dotnet test` specific, but I can not find any similar issue.

```
Running: dotnet test *.Tests.dll --settings:test.runsettings --verbosity diag -- xunit.parallelizeAssembly=true
Microsoft (R) Test Execution Command Line Tool Version 17.3.3 (x64)
Copyright (c) Microsoft Corporation.  All rights reserved.

Starting test execution, please wait...
A total of 5 test files matched the specified pattern.


[xUnit.net 00:00:00.00] xUnit.net VSTest Adapter v2.5.3.1+6b60a9e56a (64-bit .NET 6.0.24)
[xUnit.net 00:00:00.27]   Starting:    Assembly_A.Tests (parallel test collections = on, max threads = 8)
[xUnit.net 00:00:34.40]   Finished:    Assembly_A.Tests
[xUnit.net 00:00:00.00] xUnit.net VSTest Adapter v2.5.3.1+6b60a9e56a (64-bit .NET 6.0.24)
[xUnit.net 00:00:00.35]   Starting:    Assembly_B.Tests (parallel test collections = on, max threads = 8)
[xUnit.net 00:00:08.29]   Finished:    Assembly_B.Tests
[xUnit.net 00:00:00.00] xUnit.net VSTest Adapter v2.5.3.1+6b60a9e56a (64-bit .NET 6.0.24)
[xUnit.net 00:00:00.66]   Starting:    Assembly_C.Tests (parallel test collections = on, max threads = 8)
[xUnit.net 00:06:34.47]   Finished:    Assembly_C.Tests
[xUnit.net 00:00:00.00] xUnit.net VSTest Adapter v2.5.3.1+6b60a9e56a (64-bit .NET 6.0.24)
[xUnit.net 00:00:00.65]   Starting:    Assembly_D.Tests (parallel test collections = on, max threads = 8)
[xUnit.net 00:01:43.43]   Finished:    Assembly_D.Tests
[xUnit.net 00:00:00.00] xUnit.net VSTest Adapter v2.5.3.1+6b60a9e56a (64-bit .NET 6.0.24)
[xUnit.net 00:00:05.88]   Starting:    Assembly_E.Tests (parallel test collections = on, max threads = 8)
[xUnit.net 00:39:42.62]   Finished:    Assembly_E.Tests

```

My xunit.runner.json

```json
{
    "diagnosticMessages": true,
    "longRunningTestSeconds": 10,
    "shadowCopy": false,
    "parallelizeAssembly": true,
    "parallelizeTestCollections": true,
    "maxParallelThreads": 8,
    "internalDiagnosticMessages": false
}
```

My runsettings file
```xml
<?xml version="1.0" encoding="utf-8" ?>
<RunSettings>
  <xUnit>
    <ParallelizeAssembly>true</ParallelizeAssembly>
  </xUnit>
...
```

**Comment by bradwilson** [comment]
I wrote an extensive description of how VSTest works w/ respect to parallelism and how the `parallelizeAssembly` configuration element is completely ignored when using VSTest: https://github.com/xunit/visualstudio.xunit/issues/191#issuecomment-1763975518

You're invoking `dotnet test` with a list of assemblies, which basically becomes a direct call to `dotnet vstest` (without the benefit of being able to use `dotnet vstest`'s command line switches). In my opinion you have two choices to make everything run in parallel:

- Stop passing assemblies to `dotnet test` and let it just find and run all your test projects naturally. When invoked this way, it will run everything in parallel (there's actually no way to _prevent_ it from running them in parallel).
- Call `dotnet vstest` instead of `dotnet test` so that you can also pass the `--parallel` switch to it to tell it to run everything in parallel.
